### DISCUSSION TITLE
Controller is called locally but not when I deploy the operator

### DISCUSSION THREAD

**Discussion by wmeints** [dotnet/dotnet-operator-sdk#503] [Q&A]
Hi, I'm running into a pretty weird issue with the operator code. When I run my operator locally it calls the controller when my custom resource is created or modified. However, when I deploy the operator to my cluster, it's no longer being called. Any ideas what's happening?

**Comment by buehler** [comment]
Hey @wmeints 
Are the CRD's properly installed? Do you have any logs?

**Reply by wmeints** [reply]
The log is empty, apart from a message telling me that the validation hook is active and return true.
I checked the state in the cluster and my CRD is registered and looks correct.

I will try to regenerate the CRD and the rbac manifests to make sure those are correct. The rbac looks rather funky.

**Reply by buehler** [reply]
The rbac is only "more or less" responsible for a functioning controller. If your controller does not get called at all, something else is broken.

The Kubernetes API will always create and update your resources. However, a running controller can watch the resources and reconcile them.

Do you run the controller namespaced? What happens if you remove the validator and only use normal controller reconciliation?

**Reply by laurentlevi** [reply]
Hi , 

I have the exact same problem.
Controller is called from Visual Studio but not called when the operator is deployed on Kubernetes.
Not sure how I can Debug this ?
